### PR TITLE
Fix: Improve performance of streaming message handling

### PR DESCRIPTION
**Pull request by ThisIsMissEm**
This should hopefully reduce almost all our GC pressure by only calculating the searchable text contents once per message from Redis, and only if absolutely needed, rather than calculating it for each subscriber to the topic in Redis (each connected client). This means that if we have 5 clients connected on the public timeline, then the text content will be extracted via JSDOM just once, rather than 5 times.

It's similar to the change in #25278, but for the searchableTextContents which is used by filtering. Calculating the searchable text contents is easily one of the most expensive operations that streaming does, as it needs to parse a string to a DOM tree, then find all the textContent's values of each node in that DOM tree, in order to return back a string, only to discard the entire DOM tree structure.

Ideally we'd move this into the rails code, but doing that will require changes to the presenters/serializers, so this is the more isolated approach.

### CODE REVIEW

File: streaming/index.js [hunk 1]
```diff
@@ -165,6 +166,68 @@ const redisConfigFromEnv = (env) => {
...
+  'off',
+  'OFF',
+];
+
+/**
+ * @param {any} value
+ * @returns {boolean}
+ */
+const isTruthy = value =>
+  value && !FALSE_VALUES.includes(value);
```
Comment (sorin-davidoi) [inline streaming/index.js:186]: `const FALSE_VALUES = new Set([...])` and `FALSE_VALUES.has(value)` avoids iterating over the entire array.

File: streaming/index.js [hunk 2]
```diff
@@ -208,6 +338,7 @@ const startServer = async () => {
    * @param {string} channel
    */
   const onRedisMessage = (message, channel) => {
+    const isPublicChannel = PUBLIC_CHANNELS.includes(channel.slice('timeline:'.length));
```
Comment (sorin-davidoi) [inline streaming/index.js:341]: Same comment as for `FALSE_VALUES`, `PUBLIC_CHANNELS` could be a `Set` to speed-up the check.

**Review by sorin-davidoi — COMMENTED** [review]
Two small performance-related suggestions.

**Comment by ThisIsMissEm** [comment]
> Two small performance-related suggestions.

Yeah, I don't disagree with those suggestions @sorin-davidoi, but in this case I'm just lifting code up, rather than changing implementation too much

**Comment by ThisIsMissEm** [comment]
I'm closing this in favour of doing the logic for calculating searchable text in the ruby side, which is the far cleaner way to remove the JSDOM cost & dependency.

**Pull request closed without merging**